src/nfts/NftsInfoCarousel.tsx

```diff
@@ -0,0 +1,369 @@
+              TAG,
+              `Error loading Nft thumbnail image for Nft contractAddress: ${nft.contractAddress} tokenId: ${nft.tokenId}`
+            )
+          }}
+          onLoadEnd={() => {
+            setLoading(false)
+          }}
+          resizeMode={FastImage.resizeMode.cover}
+        >
+          {loading && <ThumbnailImagePlaceholder />}
+        </FastImage>
+      )}
+    </Touchable>
+  )
+}
+interface MainImagePlaceholderProps {
+  height: number
+}
+
+function MainImagePlaceholder({ height }: MainImagePlaceholderProps) {
+  return (
+    <SkeletonPlaceholder
+      borderRadius={8}
+      backgroundColor={colors.gray2}
+      highlightColor={colors.white}
+    >
+      <View
+        style={{
+          height,
+          width: variables.width,
+          zIndex: -1,
+        }}
+      />
+    </SkeletonPlaceholder>
+  )
+}
+
+interface NftImageCarouselProps {
+  nfts: Nft[]
+  handleOnPress: (nft: Nft) => void
+  activeNft: Nft
+}
+
+function NftImageCarousel({ nfts, handleOnPress, activeNft }: NftImageCarouselProps) {
+  return (
+    <View style={styles.nftImageCarouselContainer}>
+      <ScrollView
+        showsHorizontalScrollIndicator={false}
+        horizontal={true}
+        contentContainerStyle={styles.carouselScrollViewContentContainer}
+        style={styles.nftImageCarousel}
+        testID="NftsInfoCarousel/NftImageCarousel"
+      >
+        {nfts.map((nft) => {
+          return (
+            <View
+              key={`${nft.contractAddress}-${nft.tokenId}`}
+              style={styles.nftThumbnailSharedContainer}
+            >
+              <NftThumbnail
+                onPress={() => handleOnPress(nft)}
+                // Use contractAddress and tokenId for a unique key
+                isActive={
+                  `${activeNft.contractAddress}-${activeNft.tokenId}` ===
+                  `${nft.contractAddress}-${nft.tokenId}`
+                }
+                nft={nft}
+              />
+            </View>
+          )
+        })}
+      </ScrollView>
+    </View>
+  )
+}
+
+type Props = NativeStackScreenProps<StackParamList, Screens.NftsInfoCarousel>
+
+export default function NftsInfoCarousel({ route }: Props) {
+  const { nfts } = route.params
+  const [activeNft, setActiveNft] = useState(nfts[0] ?? null)
+  const [isLoading, setIsLoading] = useState(true)
+  const [scaledHeight, setScaledHeight] = useState(DEFAULT_IMAGE_HEIGHT)
+  const { t } = useTranslation()
+
+  function pressExplorerLink() {
+    navigate(Screens.WebViewScreen, {
+      uri: `${networkConfig.celoExplorerBaseTokenUrl}${activeNft.contractAddress}/instance/${activeNft.tokenId}/metadata`,
+    })
+  }
+
+  function handleThumbnailPress(nft: Nft) {
+    setActiveNft(nft)
+  }
+
+  // Display error screen no Nfts provided
+  if (nfts.length === 0) {
+    return <NftsLoadErrorScreen />
+  }
+
+  return (
+    <SafeAreaView edges={['top']} style={styles.safeAreaView} testID="NftsInfoCarousel">
+      <ScrollView>
```

Comment:
Do we account for the bottom inset needed to fully view and interact with the last element in the ScrollView on devices with the home bar?

Reply:
@Jean would this be an Android device with a bottom bar?

Reply:
Just on iOS with the home bar.

Reply:
There is some marginBottom added to the explorer link via the `sectionContainerLast` style so that it displays above the bottom inset when scrolled similar to the readMoreButton in `src/exchange/CeloNewsFeed.tsx`. Is there a better way to do this?

### iPhone 16 Long Content
https://github.com/valora-inc/wallet/assets/26950305/ff2352c4-6619-4112-93dd-61493afddfbc

### iPhone 16 Short Content

https://github.com/valora-inc/wallet/assets/26950305/d308e02d-3989-4cd2-855a-7eeb93da7319